components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/resources/push-auth.yaml

```diff
@@ -0,0 +1,590 @@
+          description: Forbidden
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - admin
+
+  /{userId}/push-auth/devices:
+    get:
+      description: |
+        This API is used by admins to get a list of devices registered under a user.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/manage/identity/user/push_device_mgt/list <br/>
+          <b>Scopes required:</b>
+        * internal_push_device_list
+      summary: |
+        Get user's registered device list.
+      parameters:
+        - name: "userId"
+          in: "path"
+          description: "ID of user"
+          required: true
+          schema:
+            type: "string"
+      responses:
+        200:
+          description: List of registered devices of the user
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AllDevices'
+        401:
+          description: Unauthorized
+        403:
+          description: Forbidden
+        404:
+          description: Not Found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - admin
+
+components:
+  schemas:
+    DiscoveryDataDTO:
+      type: object
+      description: "Device registration information"
+      properties:
+        did:
+          description: "Device ID"
+          type: string
+          example: "b03f90c9-6723-48f6-863b-a35f1ac77f57"
+        un:
+          description: "Username"
+          type: string
+          example: "admin"
+        fn:
+          description: "First name"
+          type: string
+          example: "John"
+        ln:
+          description: "Last name"
+          type: string
+          example: "Doe"
+        td:
+          description: "Tenant domain/Organisation"
+          type: string
+          example: "mainDomain"
+        hst:
+          description: "Base URL"
+          type: string
+          format: hostname
+          example: "https://localhost:9443/"
+        bp:
+          description: "Base path for the me endpoints"
+          type: string
+          example: "t/{tenant_context}/api/users/v1/me"
+        re:
+          description: "Registration endpoint"
+          type: string
+          format: url
+          example: "/push-auth/devices"
+        ae:
+          description: "Authentication endpoint"
+          type: string
+          format: url
+          example: "push-auth/authenticate"
+        rde:
+          description: "Remove device endpoint"
+          type: string
+          format: url
+          example: "push-auth/devices/remove"
```

Comment:
Mark these optional. Only the hostname should be there as others will not change. Open an issue